### PR TITLE
Added information branch to graphRestrictions. Fixes #1450

### LINKED ISSUE #1450: Include 'Information entity' branch as potential ontology terms for file_content_description
I would like to request an update to the module/ontology/file_content_ontology.json schema

**What should the change/update be?**
This would be a change to the ontology graph restrictions. Currently, we are restricted to ontology terms which are a from obo:edam, which are a subclass of data:0006 (Data). We have recently added a new branch 'Information entity' (IAO:0000030) which has useful file content descriptions. 

We should expand the ontology term restrictions for file_content_description to include 'Information entity' (IAO:0000030). 

This update constitutes a minor change to the schema(s) it affects.

**What new field(s) need to be changed?**
The module/ontology/file_content_ontology.json schema field needs to be changed

ontology: {
description: "An ontology term identifier in the form prefix:accession.",
type: "string",
Required: no
Examples: EFO:0010198, cell barcode  

This will allow appropriate ontology terms to be used for different file formats (e.g. UMI barcode, cell barcode, sample barcode)

### PR DESCRIPTION
**Pull request by ESapenaVentura**
Schema update priority ticket with explanation on the change: https://app.zenhub.com/workspaces/operations-5fa2d8f2df78bb000f7fb2b5/issues/ebi-ait/hca-ebi-wrangler-central/695

### Release notes

For module/ontology/file_content_ontology schema:
- Changed restrictions to allow `Information` branch to the allowed terms.

### CODE REVIEW

File: json_schema/update_log.csv [hunk 1]
```diff
@@ -1 +1,2 @@
 Schema,Change type,Change message,Version,Date
+module/ontology/file_content_ontology,minor,"Added information branch to graphRestrictions. Fixes #1450",,
```
Comment (hannes-ucsc) [inline json_schema/update_log.csv:2]:
```suggestion
module/ontology/file_content_ontology,minor,"Added information entity branch to graphRestrictions. Fixes #1450",,
```

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
I think I need some advice on how to produce an exhaustive list of all ontologies used in HCA metadata for 

https://github.com/DataBiosphere/azul/issues/3636

Ultimately I need all the OWL files so I can index them.

**Comment by ESapenaVentura** [comment]
@hannes-ucsc thanks for the review! about your request of the exhaustive list, we have an endpoint that returns all the latest schemas in ingest (`https://api.ingest.archive.data.humancellatlas.org/schemas/search/latestSchemas`) that you can query to obtain the latest schemas

I have written a small snippet to show how you could retrieve a list of the ontology schemas:

```python3
import requests as rq


def get_latest_schemas():
    url = "https://api.ingest.archive.data.humancellatlas.org/schemas/search/latestSchemas?size=200"
    response = rq.get(url)
    response.raise_for_status()
    return response.json().get('_embedded', {}).get('schemas', [])


def filter_ontology_schemas(schema_list):
    return [schema for schema in schema_list if schema.get('domainEntity', "") == "ontology"]


def get_schemas(schema_list):
    actual_schema_list = []
    for schema in schema_list:
        r = rq.get(schema.get('_links', {}).get('json-schema', {}).get('href'))
        r.raise_for_status()
        actual_schema_list.append(r.json())
    return actual_schema_list


def main():
    latest_schema_list = get_latest_schemas()
    list_ontology_schemas_pointers = filter_ontology_schemas(latest_schema_list)
    latest_ontology_schemas = get_schemas(list_ontology_schemas_pointers)
    return latest_ontology_schemas


if __name__ == '__main__':
    main()
```

With that, you could extract from the restrictions (`properties.ontology.graph_restriction`) the ontologies used (`ontologies`) or, if you want to be even more specific, the terms allowed (`classes`).

Once you know that, it shouldn't be too difficult to get the OWL files, although I am not entirely sure of the path to follow for that. There is some documentation on the OLS here https://www.ebi.ac.uk/ols/docs/developer

**Comment by hannes-ucsc** [comment]
> With that, you could extract from the restrictions (properties.ontology.graph_restriction) the ontologies used (ontologies) or, if you want to be even more specific, the terms allowed (classes).

That's the part I am struggling with.

**Comment by hannes-ucsc** [comment]
As in how do I get from the string https://github.com/HumanCellAtlas/metadata-schema/blob/ff27788edc635f6e631a64b3fcda63f54fc6fe7b/json_schema/module/ontology/contributor_role_ontology.json#L33 to the URL of a REST endpoint that returns a link to the ontology?

**Comment by hannes-ucsc** [comment]
Should I just ignore the `obo` bit and expect that part after that (`efo`) can be used to search for the ontology by "short name" at https://www.ebi.ac.uk/ols/ontologies/iao?

**Comment by ESapenaVentura** [comment]
hmmm... I am asking other people so please take this with a grain of salt, but if you use the second part of the identifier (e.g. from obo:efo, only "efo") as an ontology_id, I think you can retrieve the ontology from the following endpoint in OLS:
`http://www.ebi.ac.uk/ols/api/ontologies/{ontology_id}`

This request should give you the ontology, and within the `config` field I think you can find the OWL file to download.

I will further check with other member of the teams to see if we have implemented some easier way to retrieve the OWL files!

**Review by kbergin — APPROVED** [review]

**Review by ruchim — APPROVED** [review]
lgtm!

**Review by amnonkhen — APPROVED** [review]
LGTM

**Review by NoopDog — APPROVED** [review]

**Pull request merged**